src/main/java/org/opensearch/neuralsearch/plugin/NeuralSearch.java

```diff
@@ -71,4 +76,11 @@ public List<QuerySpec<?>> getQueries() {
             new QuerySpec<>(NeuralQueryBuilder.NAME, NeuralQueryBuilder::new, NeuralQueryBuilder::fromXContent)
         );
     }
+
+    @Override
+    public Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
+        final MachineLearningNodeClient machineLearningNodeClient = new MachineLearningNodeClient(parameters.client);
+        mlCommonsClientAccessor = new MLCommonsClientAccessor(machineLearningNodeClient);
```

Comment:
We need to create the MLCommonsClientAccessor in the getProcessors method since this method executes before `createComponents`, check this line: https://github.com/opensearch-project/OpenSearch/blob/e44158d4d10d4f8905895ffa50bf9398b8550667/server/src/main/java/org/opensearch/node/Node.java#L515, and this line: https://github.com/opensearch-project/OpenSearch/blob/e44158d4d10d4f8905895ffa50bf9398b8550667/server/src/main/java/org/opensearch/node/Node.java#L711.